pyanaconda/ui/lib/subscription.py

```diff
@@ -287,17 +332,21 @@ def register_and_subscribe(payload, progress_callback=None, error_callback=None,
         error_callback(_("Registration failed due to insufficient credentials."))
         return
 
-    # try to attach subscription
-    log.debug("subscription thread: attempting to auto attach an entitlement")
-    progress_callback(SubscriptionPhase.ATTACH_SUBSCRIPTION)
-    task_path = subscription_proxy.AttachSubscriptionWithTask()
-    task_proxy = SUBSCRIPTION.get_proxy(task_path)
-    try:
-        task.sync_run_task(task_proxy)
-    except SubscriptionError as e:
-        log.debug("subscription thread: failed to attach subscription: %s", e)
-        error_callback(str(e))
-        return
+    if sca_enabled:
+        # skip auto-attach as it is not necessary in Simple Content Access mode
+        log.debug("subscription thread: auto attach skipped due to Simple Content Access mode")
+    else:
+        # not in SCA mode - try to auto attach subscription
+        log.debug("subscription thread: attempting to auto attach an entitlement")
+        progress_callback(SubscriptionPhase.ATTACH_SUBSCRIPTION)
+        task_path = subscription_proxy.AttachSubscriptionWithTask()
```

Comment:
So, the UI actually doesn't need to know anything about the Simple Content Access mode. It could be all handled in the `AttachSubscriptionTask` class that would internally check the mode and skip the rest. Since we want to keep the DBus API as simple as possible, I definitely prefer that solution.